ToolBox/CoinApi/CoinApiSymbolMapper.cs

```diff
@@ -137,27 +141,38 @@ public string GetExchangeId(string market)
         private void LoadSymbolMap(string[] exchangeIds)
         {
             var list = string.Join(",", exchangeIds);
+            var json = string.Empty;
 
-            using (var wc = new WebClient())
+            if (_useLocalSymbolList && _coinApiSymbolsListFile.Exists && _coinApiSymbolsListFile.CreationTimeUtc.Date < DateTime.UtcNow.Date.AddDays(-1))
```

Comment:
Am I reading this correct? We want to use the local file if it is _older_ than yesterday? Also, using a file time to determine the age of data may not be the best -- wondering if it would be better to put the time the data was generated into the file instead? Consider a file being a week old and then we copy it to a new machine and it would show as new (methinks?).

Reply:
Aside from the inverted file date check, a couple of alternatives to using the file timestamp:
1. adding the date in the filename, e.g. `CoinApiSymbols.yyyyMMdd.json`
2. reading the last date from the data itself (in the `data_end` field), this is the file format:
```
[
  {
    "symbol_id": "COINBASE_SPOT_BTC_USD",
    "exchange_id": "COINBASE",
    "symbol_type": "SPOT",
    "asset_id_base": "BTC",
    "asset_id_quote": "USD",
    "data_start": "2015-01-14",
    "data_end": "2019-05-29",
    "data_quote_start": "2015-05-17T00:51:36.2870000Z",
    "data_quote_end": "2019-05-29T22:57:31.6924043Z",
    "data_orderbook_start": "2015-05-17T00:51:36.2870000Z",
    "data_orderbook_end": "2019-05-29T19:02:17.4261189Z",
    "data_trade_start": "2015-01-14T16:07:05.0000000Z",
    "data_trade_end": "2019-05-29T23:05:46.5590000Z",
    "volume_1hrs": 421.426615810,
    "volume_1hrs_usd": 3681568.97,
    "volume_1day": 12742.682746710,
    "volume_1day_usd": 111319654.79,
    "volume_1mth": 603712.812420610,
    "volume_1mth_usd": 5274015151.04
  },
  ...
]
```

Reply:
My bad, not enough coffee that afternoon.

Reply:
Thinking more about it, I'm wondering if we even need the file date check 🤔 

When the `CoinApiSymbolMapper` is used in the `CoinApiDataQueueHandler` or in an occasional converter run, there will only be a single instance, so API rate limits are not an issue.

When it's used for massively parallel processing with the `CoinApiDataConverter`, the date check could return false for all processes and we wouldn't guarantee rate limits won't be hit. Perhaps for these scenarios, the file could be pre-downloaded (once a day) and the symbol mapper would just read the file with no date checks?

``` C#
if (_useLocalSymbolList)
{
    if (!_coinApiSymbolsListFile.Exists) 
    {
        throw new Exception($"File not found: {_coinApiSymbolsListFile.FullName}");
    }
}
```